jwst/ami/oifits.py

```diff
@@ -134,50 +301,53 @@ def make_oifits(self):
             self.e_visphi = np.zeros(self.visphi.shape)
             self.closure_phases = self.closure_phases.T
             self.e_cp = np.zeros(self.closure_phases.shape)
+            self.t3amp = self.t3_amplitudes.T
+            self.e_t3amp = np.zeros(self.t3amp.shape)
+            self.q4phi = self.q4_phases.T
+            self.e_q4phi = np.zeros(self.q4_phases.shape)
             self.camp = self.closure_amplitudes.T
             self.e_camp = np.zeros(self.camp.shape)
             self.pist = self.pistons.T
             self.e_pist = np.zeros(self.pist.shape)
 
-        # apply sigma-clipping to uncertainties
-        # sigma_clipped_stats returns mean, median, stddev. nsigma=3, niters=5
-        elif self.method == "median":
-            _, self.vis2, self.e_vis2 = sigma_clipped_stats(self.fringe_amplitudes_squared, axis=0)
-            _, self.visamp, self.e_visamp = sigma_clipped_stats(self.fringe_amplitudes, axis=0)
-            _, self.visphi, self.e_visphi = sigma_clipped_stats(self.fringe_phases, axis=0)
-            _, self.closure_phases, self.e_cp = sigma_clipped_stats(self.closure_phases, axis=0)
-            _, self.camp, self.e_camp = sigma_clipped_stats(self.closure_amplitudes, axis=0)
-            _, self.pist, self.e_pist = sigma_clipped_stats(self.pistons, axis=0)
-
-        else:  # take the mean
-            self.vis2, _, self.e_vis2 = sigma_clipped_stats(self.fringe_amplitudes_squared, axis=0)
-            self.visamp, _, self.e_visamp = sigma_clipped_stats(self.fringe_amplitudes, axis=0)
-            self.visphi, _, self.e_visphi = sigma_clipped_stats(self.fringe_phases, axis=0)
-            self.closure_phases, _, self.e_cp = sigma_clipped_stats(self.closure_phases, axis=0)
-            self.camp, _, self.e_camp = sigma_clipped_stats(self.closure_amplitudes, axis=0)
-            self.pist, _, self.e_pist = sigma_clipped_stats(self.pistons, axis=0)
+
+        elif self.method == "mean":
+            self.vis2, self.e_vis2, self.visamp, self.e_visamp, self.visphi, self.e_visphi, self.closure_phases, self.e_cp, self.t3amp, self.e_t3amp, self.camp, self.e_camp, self.q4phi, self.e_q4phi, self.pist, self.e_pist =  self.average_observables(np.mean)
+
+        else:  # take the median
+            self.vis2, self.e_vis2, self.visamp, self.e_visamp, self.visphi, self.e_visphi, self.closure_phases, self.e_cp, self.t3amp, self.e_t3amp, self.camp, self.e_camp, self.q4phi, self.e_q4phi, self.pist, self.e_pist =  self.average_observables(np.median)
```

Comment:
These lines are very long - we should add some newlines to make them more readable
```suggestion
            (self.vis2, self.e_vis2, self.visamp, self.e_visamp, 
             self.visphi, self.e_visphi, self.closure_phases, self.e_cp, 
             self.t3amp, self.e_t3amp, self.camp, self.e_camp, self.q4phi, 
             self.e_q4phi, self.pist, self.e_pist) = self.average_observables(np.mean)

        else:  # take the median
            (self.vis2, self.e_vis2, self.visamp, self.e_visamp, 
             self.visphi, self.e_visphi, self.closure_phases, self.e_cp, 
             self.t3amp, self.e_t3amp, self.camp, self.e_camp, self.q4phi, 
             self.e_q4phi, self.pist, self.e_pist) = self.average_observables(np.median)
```